sectionproperties/pre/sections.py

```diff
@@ -419,84 +458,181 @@ def calculate_perimeter(self):
+    if r == 0:
+        points.append(pt)
+        return
+
+    if ccw:
+        mult = 1
+    else:
+        mult = -1
+
+    # calculate radius of points
+    for i in range(n):
+        # determine angle
+        t = theta + mult * i * 1.0 / max(1, n - 1) * np.pi * 0.5
+
+        x = pt[0] + r * np.cos(t)
+        y = pt[1] + r * np.sin(t)
+        points.append([x, y])
+    return points
+
+# class CustomSection(Geometry):
+#     """Constructs a cross-section from a list of points, facets, holes and a user specified control
+#     point.
+
+#     :param points: List of points *(x, y)* defining the vertices of the cross-section
+#     :type points: list[list[float, float]]
+#     :param facets: List of point index pairs *(p1, p2)* defining the edges of the cross-section
+#     :type facets: list[list[int, int]]
+#     :param holes: List of points *(x, y)* defining the locations of holes within the cross-section.
+#         If there are no holes, provide an empty list [].
+#     :type holes: list[list[float, float]]
+#     :param control_points: A list of points *(x, y)* that define different regions of the
+#         cross-section. A control point is an arbitrary point within a region enclosed by facets.
+#     :type control_points: list[list[float, float]]
+#     :param shift: Vector that shifts the cross-section by *(x, y)*
+#     :type shift: list[float, float]
+#     :param perimeter: List of facet indices defining the perimeter of the cross-section
+#     :vartype perimeter: list[int]
+
+#     The following example creates a hollow trapezium with a base width of 100, top width of 50,
+#     height of 50 and a wall thickness of 10. A mesh is generated with a maximum triangular area of
+#     2.0::
+
+#         import sectionproperties.pre.sections as sections
+
+#         points = [[0, 0], [100, 0], [75, 50], [25, 50], [15, 10], [85, 10], [70, 40], [30, 40]]
+#         facets = [[0, 1], [1, 2], [2, 3], [3, 0], [4, 5], [5, 6], [6, 7], [7, 4]]
+#         holes = [[50, 25]]
+#         control_points = [[5, 5]]
+#         perimeter = [0, 1, 2, 3]
+
+#         geometry = sections.CustomSection(
+#             points, facets, holes, control_points, perimeter=perimeter
+#         )
+#         mesh = geometry.create_mesh(mesh_sizes=[2.0])
+
+#     ..  figure:: ../images/sections/custom_geometry.png
+#         :align: center
+#         :scale: 75 %
+
+#         Custom section geometry.
+
+#     ..  figure:: ../images/sections/custom_mesh.png
+#         :align: center
+#         :scale: 75 %
+
+#         Mesh generated from the above geometry.
+#     """
+
+#     def __init__(self, points, facets, holes, control_points, shift=[0, 0], perimeter=[]):
+#         """Inits the CustomSection class."""
+
+#         super().__init__(control_points, shift)
+
+#         self.points = points
+        # self.facets = facets
+        # self.holes = holes
+        # self.perimeter = perimeter
+
+        # self.shift_section()
+
+
+def rectangular_section(b, d):
```

Comment:
Given that this will be an "overhaul" of the preprocessing (which will include some new methods in `Geometry` and remove old methods), I think it would be a 2.0.x release, indicating breaking changes. However, it would be great to include the DXF features in the release, too. It would *feel* like a big upgrade in functionality/usability to go along with any breaking changes.